addons/web/static/src/js/fields/signature.js

```diff
@@ -230,19 +231,13 @@ var FieldBinarySignature = AbstractFieldBinary.extend({
                 noInputName: true,
             };
 
-            if (this.nodeOptions.full_name) {
-                var $search = $("[name='" + this.nodeOptions.full_name + "']");
-                // If full_name correspond to field in read-only mode
-                var signName = $($search).text();
-
-                // If full_name correspond to field to fill
-                if (!signName && $($search)[0]) {
-                    signName = $($search)[0].value;
-                }
-                if (!signName) {
-                    // If full_name is in the recordData. Last option because this value might not be the last value.
-                    signName = this.recordData[this.nodeOptions.full_name] || '';
-                }
+            if (this.nodeOptions.full_name && this.fields[this.nodeOptions.full_name]) {
+                var signName;
+                if (this.fields[this.nodeOptions.full_name].type === 'many2one' && this.recordData[this.nodeOptions.full_name]) {
+                    signName = this.recordData[this.nodeOptions.full_name].data.display_name || '';
```

Comment:
is `|| ''` really useful? If `display_name` is not set, then during assignation (see l241), `defaultName` will have the same value.

Reply:
Right. Removed it.